plugins/SitesManager/API.php

```diff
@@ -394,6 +394,22 @@ public function getSitesWithAdminAccess($fetchAliasUrls = false, $pattern = fals
         return $sites;
     }
 
+    /**
+     * Returns the messages to warn users on site deletion.
+     *
+     * @throws Exception if the website ID doesn't exist or the user doesn't have access to it
+     * @param int $idSite
+     * @return string messages to warn users
+     */
+    public function getMessagesToWarnOnSiteRemoval($idSite)
+    {
+        $messages = [Piwik::translate('SitesManager_DeleteSiteExplanation')];
+        Piwik::checkUserHasViewAccess($idSite);
+        Piwik::postEvent('SitesManager.getMessagesToWarnOnSiteRemoval', [$idSite, &$messages]);
+
+        return implode("<br>", $messages);
```

Comment:
We should return the array here instead, and let the consumer decide how to put that together.

```suggestion
        return $messages;
```